tests/baselines/reference/jsDeclarationsFunctions.js

```diff
@@ -173,6 +173,6 @@ export function g(a: {
 declare function hh(a: {
     x: string;
 }, b: {
-    y: typeof b;
+    y: typeof import(".").b;

```

Comment:
This is suspicious; the code above says defines param `b`'s type as `{y: typeof b}` , which is self referential and surely wrong.